File: ItemBookingApp-API/Domain/Models/OrderAggregate/Order.cs
namespace ItemBookingApp_API.Domain.Models.OrderAggregate
{
    public class Order
    {
        public Order()
        {

        }
        public Order(IReadOnlyList<OrderItem> orderItems, string borrowerEmail, Address shipToAddress, 
            DeliveryMethod deliveryMethod, 
            decimal subTotal)
        {
            //Id = id;
            BorrowerEmail = borrowerEmail;
            ShipToAddress = shipToAddress;
            DeliveryMethod = deliveryMethod;
            OrderItems = orderItems;
            SubTotal = subTotal;
        }

        public int Id { get; set; }
        public string BorrowerEmail { get; set; } = string.Empty;
        public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;

        public Address ShipToAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }

        public IReadOnlyList<OrderItem> OrderItems { get; set; }

        public decimal SubTotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;

        public string PaymentItentId { get; set; }

        public decimal GetTotal () 
        {
            return SubTotal + DeliveryMethod.Price;
        }
    }
}

File: ESS-SPA/src/app/entities/models/order.ts
import {IAddress} from './address';

export interface IOrderToCreate {
    basketId: number;
    deliveryMethodId: number;
    shipToAddress: IAddress
}


export interface IOrder {
    id: number;
    borrowerEmail: string;
    orderDate: string;
    shipToAddress: IAddress;
    bookingInformation: IBookingInformation;
    deliveryMethod: string;
    shippingPrice: number;
    orderItems: IOrderItem[];
    subTotal: number;
    total: number;
    status: string;    
}

export interface IBookingInformation {
    bookingStatus: string;
    startDate: string;
    endDate: string;
    returnedDate: string;
}

export interface IOrderItem {
    itemId: number;
    itemName: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

export enum OrderPaymentStatus{
    Pending,
    PaymentReceived,
    PaymentFailed,
}

export enum OrderBookingStatus{
    Pending,
    Approved,
    Closed
}